### PR TITLE
Cleanup toolbar design

### PR DESCRIPTION
**Pull request by ThomasOliverKimble**
**Description**
New webotsView toolbar design for stream, animation and scenes, new icons and animations, style mods and bug fixes.

animation
scene
floatingWindows
worldSelect
robotRindowSelect

### CODE REVIEW

File: resources/web/wwi/FloatingIde.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+export default class FloatingIDE extends FloatingWindow {
+  constructor(parentNode, name, url) {
+    super(parentNode, name, url);
+
+    this.headerText.innerHTML = 'IDE';
+
+    this.frame.id = this.name + '-robot-window';
+    const elements = url.split('/').filter(element => element);
+    let portString = elements[elements.length - 1];
+    let port = parseFloat(portString) + 500;
```
Comment (BenjaminDeleze) [inline resources/web/wwi/FloatingIde.js:12]:
```suggestion
    const portString = elements[elements.length - 1];
    const port = parseFloat(portString) + 500;
```
My bad, I forgot to make them `const`

File: resources/web/wwi/FloatingWindow.js [hunk 2]
```diff
@@ -29,21 +29,49 @@ export default class FloatingWindow {
...
+      let fw = document.getElementById(name)
+      let maxHeight = fw.parentNode.offsetHeight - fw.offsetTop;
+      let maxWidth = fw.parentNode.offsetWidth - fw.offsetLeft;
+      fw.style.maxHeight = maxHeight.toString() + 'px';
+      fw.style.maxWidth = maxWidth.toString() + 'px';
+      fw.lastElementChild.style.pointerEvents = 'none';
+    }
+
+    function restorePointerEvents(name) {
+      let fw = document.getElementById(name);
```
Comment (BenjaminDeleze) [inline resources/web/wwi/FloatingWindow.js:50]: let to const?

File: resources/web/wwi/FloatingWindow.js [hunk 3]
```diff
@@ -64,21 +92,21 @@ export default class FloatingWindow {
...
 
-  dragElement(fw) {
+  _dragElement(fw) {
     let pos1 = 0, pos2 = 0, pos3 = 0, pos4 = 0;
-    let pageHeight = fw.parentNode.parentNode.parentNode.offsetHeight;
-    let containerHeight = fw.parentNode.parentNode.offsetHeight;
-    let containerWidth = fw.parentNode.parentNode.offsetWidth;
+    let pageHeight = fw.parentNode.parentNode.offsetHeight;
+    let containerHeight = fw.parentNode.offsetHeight;
+    let containerWidth = fw.parentNode.offsetWidth;
```
Comment (BenjaminDeleze) [inline resources/web/wwi/FloatingWindow.js:99]: const?

File: resources/web/wwi/Toolbar.js [hunk 4]
```diff
@@ -423,11 +481,19 @@ export default class Toolbar {
...
-      this.robotWindows.forEach((rw) => this._changeRobotWindowVisibility(rw.getID()));
-    this.robotWindowButton.addEventListener('mouseup', _ => this._changeRobotWindowPaneVisibility());
+      this.robotWindows.forEach((rw) => this._changeFloatingWindowVisibility(rw.getId()));
+    this.robotWindowButton.addEventListener('mouseup', _ => this._changeRobotWindowPaneVisibility(_));
     document.addEventListener('keydown', this.keydownRefW = _ => this._robotWindowPaneKeyboardHandler(_, false));
   }
 
+  _fullscreenChangeHeight(floatingWindows) {
+    floatingWindows.forEach ((fw) => {
+      let maxHeight = this.parentNode.offsetHeight - fw.getPosition()[0];
```
Comment (BenjaminDeleze) [inline resources/web/wwi/Toolbar.js:491]:
```suggestion
      const maxHeight = this.parentNode.offsetHeight - fw.getPosition()[0];
```

**Review by BenjaminDeleze — CHANGES_REQUESTED** [review]
Seems good, only some style suggestions

File: resources/web/wwi/FloatingWindow.js (hunk 3 above)
Comment (ThomasOliverKimble) [reply resources/web/wwi/FloatingWindow.js:99]: These values will change if the user changes screen size. I just tested with const and it doesn't work.

File: docs/guide/web-simulation.md [hunk 5]
```diff
@@ -447,11 +447,12 @@ Moreover, the following attributes are available:
...
 * `showReset`: specify if the reset button must be displayed on the toolbar. Must be called before connect. The reset button is displayed by default.
-* `showRobotWindow`: specify if the robot window button must be displayed on the toolbar. Must be called before connect. The robot windows button is displayed by default.
+* `showRobotWindow`: specify if the robot window button must be displayed on the toolbar. Must be called before connect. The robot window button is displayed by default.
+* `showWorldSelection`: specify if the world selection button must be displayed on the toolbar. Must be called before connect. The world selection is displayed by default.
 * `showStep`: specify if the step button must be displayed on the toolbar. Must be called before connect. The step button is displayed by default.
+* `showStep`: specify if the step button must be displayed on the toolbar. Must be called before connect. The quit button is displayed by default.
 * `showQuit`: specify if the quit button must be displayed on the toolbar. Must be called before connect. The quit button is displayed by default.
```
Comment (BenjaminDeleze) [inline docs/guide/web-simulation.md:456]: The `showReload` has disappear whereas there is now two times the `showStep`

File: resources/web/wwi/Toolbar.js [hunk 6]
```diff
@@ -46,22 +47,22 @@ export default class Toolbar {
       return;
 
     this._createInfoButton();
-    this._createRestoreViewpointButton();
+    //this._createRestoreViewpointButton();
```
Comment (BenjaminDeleze) [inline resources/web/wwi/Toolbar.js:50]:
```suggestion
```

**Review by BenjaminDeleze — CHANGES_REQUESTED** [review]
Looks good, only minor remarks

File: resources/web/wwi/Toolbar.js [hunk 7]
```diff
@@ -382,22 +404,30 @@ export default class Toolbar {
     const margin = 20;
     let numCol = 0;
     let numRow = 0;
+    let ideOffset = 0;
+    if (this._view.ide)
+      ideOffset = 520;
```
Comment (omichel) [inline resources/web/wwi/Toolbar.js:409]:
```suggestion
    const ideOffset = (this._view.ide) ? 520 : 0;
```

**Review by omichel — CHANGES_REQUESTED** [review]

**Review by BenjaminDeleze — APPROVED** [review]
Looks good to me

File: resources/web/wwi/images/icons/webotsview.svg [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<svg version="1.1" id="webotsview" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
```
Comment (omichel) [inline resources/web/wwi/images/icons/webotsview.svg:1]: To be consistent with other svg names, e.g., robot-window, world-selection, etc., I would rename this one to webots-view.svg

File: resources/web/wwi/images/icons/partscreen.svg [hunk 9]
```diff
@@ -0,0 +1,10 @@
+<svg version="1.1" id="fullscreen" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
```
Comment (omichel) [inline resources/web/wwi/images/icons/partscreen.svg:1]: I would rename this one to `windowed.svg`

**Review by omichel — CHANGES_REQUESTED** [review]
It looks good to me. I only have a couple of comments regarding file names for a couple of SVG files.

**Review by omichel — APPROVED** [review]
Thank you.

**Pull request merged**